bundles/net.tourbook.photo/src/net/tourbook/photo/ImageUtils.java

```diff
@@ -59,6 +58,42 @@ public class ImageUtils {
       _prefStore.addPropertyChangeListener(_prefChangeListener);
    }
 
+   /**
+    * For images with a transparent layer, this will keep the existing
+    * transparency.
+    * Source: https://stackoverflow.com/a/63703052
+    *
+    * @param sourceImageData
+    * @param imgWidth
+    * @param imgHeight
+    * @return
+    */
+   private static ImageData copyImageTransparencyData(final ImageData sourceImageData,
+                                                      final int imgWidth,
+                                                      final int imgHeight) {
+
+      if (sourceImageData.alphaData == null) {
+         return null;
+      }
+
+      final ImageData destData = new ImageData(imgWidth, imgHeight, sourceImageData.depth, sourceImageData.palette);
+
+      destData.alphaData = new byte[destData.width * destData.height];
+      for (int destRow = 0; destRow < destData.height; destRow++) {
+
+         for (int destCol = 0; destCol < destData.width; destCol++) {
+
+            final int origRow = destRow * sourceImageData.height / destData.height;
+            final int origCol = destCol * sourceImageData.width / destData.width;
+            final int o = origRow * sourceImageData.width + origCol;
+            final int d = destRow * destData.width + destCol;
+            destData.alphaData[d] = sourceImageData.alphaData[o];

```

Comment:
When the code is not written from me (Here it's from [here](https://stackoverflow.com/a/63703052)), then I refrain from changing it because it is external and belong to someone that is not a contributor, even the name variables (you can see here that the variable names are terrible : `o` etc...) as it is from another author but if you need me to change it and optimize it then I can do it

I noticed the same thing [here ](https://github.com/mytourbook/mytourbook/blob/main/bundles/net.tourbook.common/src/net/tourbook/common/util/FileUtils.java)where the code seems to be from Adobe and it is not optimized
And also, it seems strange that there is already an adjacent `FilesUtils` [here ](https://github.com/mytourbook/mytourbook/blob/main/bundles/net.tourbook.common/src/net/tourbook/common/util/FilesUtils.java)and never understood why they were separate

Reply:
> [](https://github.com/mytourbook/mytourbook/blob/main/bundles/net.tourbook.common/src/net/tourbook/common/util/FilesUtils.java)and never understood why they were separate

You have to ask yourself why you created a separate file, the Adobe file is 10 years older

![image](https://user-images.githubusercontent.com/1283445/222642363-15379d29-7d86-45c2-b10c-71dff83ae27f.png)

Reply:
> if you need me to change it and optimize it then I can do it

I just checked this code by loading my photos and this code is not used because a photo do not have transparency data. So for now we can keep the code as it is.

It would be interresting to know how much it could be optimized but because I don't have big transparent images, I can't do it easily.

Reply:
> You have to ask yourself why you created a separate file, the Adobe file is 10 years older
> 
> ![image](https://user-images.githubusercontent.com/1283445/222642363-15379d29-7d86-45c2-b10c-71dff83ae27f.png)

What I recall is that because I saw that the file was originating from Adobe, then I didn't want to modify it per copyright purposes and also because there is only 1 function used (`consumeBOM`) and all the rest of the code is commented, which was confusing.

Reply:
> There is code in MT from others which I also don't have modified but there is also code which I've modified to better suit my needs but I always put into a comment a hint for the origin.

Ok, I could do that. Also, where is the limit where when the original code is much modified then it's so different from the original one that it could be considered new and no need to reference the original author

Reply:
> > There is code in MT from others which I also don't have modified but there is also code which I've modified to better suit my needs but I always put into a comment a hint for the origin.
> 
> Ok, I could do that. Also, where is the limit where when the original code is much modified then it's so different from the original one that it could be considered new and no need to reference the original author

There is no limit, when I do a modification in the original code then I put a reference to the original author in the header of the code. This also helps me (years) later when I want to know where is the original source.

In this source code, there is not much the same as the original but I can click on this link and can see the original

![image](https://user-images.githubusercontent.com/1283445/222878821-9055bc6b-b365-4835-bff9-ae34738be36b.png)

Reply:
> > There is code in MT from others which I also don't have modified but there is also code which I've modified to better suit my needs but I always put into a comment a hint for the origin.
> 
> Ok, I could do that.

You can do that for the next version

Reply:
Code improved in [this PR](https://github.com/mytourbook/mytourbook/pull/1042)